src/test/java/org/testng/AssertTest.java

```diff
@@ -381,6 +381,16 @@ public void testAssertSame() {
     Assert.assertSame(object2, object);
   }
 
+  @Test(description = "GITHUB-2490")
+  public void testAssertNotEqualsWithActualNull() {
+    Assert.assertNotEquals(null, new BrokenEquals());
+  }
+
+  @Test(description = "GITHUB-2490")
+  public void testAssertNotEqualsWithExpectedNull() {
+    Assert.assertNotEquals(new BrokenEquals(), null);

```

Comment:
Additional tests:
```java
  @Test
  public void testAssertNotEqualsWithActualNullExceptedEmptyString() {
    Assert.assertNotEquals(null, "");
  }

  @Test
  public void testAssertNotEqualsWithActualEmptyStringExceptedNull() {
    Assert.assertNotEquals("", null);
  }
```

And you can keep to ones with `new Object()` as non-regression tests.

Reply:
Thanks for letting me know about this. 👍